mlexchange/mlex_highres_segmentation
# Update annotations processing

This PR updates the `Annotations` class in line with the changes on the `issue94-annotationsUI` branch where annotations for each class are stored in separate `dcc.Stores`. 

Additional changes: 
- Resolves bugs in the mask creation where `image_height` and `image_width` were mismatched, resulting in errors exporting masks of certain geometry types (eg. ellipses). 
- Removes the `create_annotation_metadata()` function and moved this to the class constructor as this processing feels fundamental to the handling of these annotations. 
- Removed the unneeded `mask_playground.ipynb` notebook.

To test:
1. Create multiple annotations across different classes and different slices
2. Click "Save and export", then "Export annotations"
3. A `.json` file with metadata and zip of `.npy` files should be downloaded. Each `.npy` file should correspond to one annotated slice of the original data. If you open and plot the `.npy` files you should see that they match the annotations you drew in the app.
